server/remote_cache/hit_tracker/hit_tracker.go

```diff
@@ -379,6 +466,43 @@ func ScoreCard(ctx context.Context, env environment.Env, iid string) *capb.Score
 	return sc
 }
 
+func readResults(ctx context.Context, env environment.Env, iid string) *capb.ScoreCard {
+	c := env.GetMetricsCollector()
+	if c == nil || iid == "" {
+		return nil
+	}
+	sc := &capb.ScoreCard{}
+
+	resultIDs, err := c.SetGetMembers(ctx, resultIDsKey(iid))
+	if err != nil {
+		log.Warningf("Failed to read detailed scorecard for invocation %s: %s", iid, err)
+		return sc
+	}
+	// TODO(bduffany): remove this limit
```

Comment:
nit: add a comment explaining why this is here and that this is not expected to be hit in production usage?

Reply:
Done, also increased the limit a bit from ~10MB worth of results to ~50MB.